src/xmpp/omemo.c

```diff
@@ -23,11 +23,9 @@ omemo_devicelist_publish(GList *device_list)
     xmpp_ctx_t * const ctx = connection_get_ctx();
     xmpp_stanza_t *iq = stanza_create_omemo_devicelist_publish(ctx, device_list);
 
-    Jid *jid = jid_create(connection_get_fulljid());
-    if (caps_jid_has_feature(jid->barejid, XMPP_FEATURE_PUBSUB_PUBLISH_OPTIONS)) {
+    if (connection_supports(XMPP_FEATURE_PUBSUB_PUBLISH_OPTIONS)) {
```

Comment:
`connection_supports` is fed with responses from `iq_disco_info_request_onconnect`.
But `omemo_devicelist_publish` is used from `omemo_on_connect` which is called in `sv_ev_login_account_success` so just before  `iq_disco_info_request_onconnect` see https://github.com/boothj5/profanity/blob/f7cb40ecf0b53a706875f1223dc99cf6481c3618/src/xmpp/session.c#L287-L320

@pasis @boothj5 any idea how this should be handled?